addons/payment_stripe/controllers/main.py

```diff
@@ -11,52 +14,46 @@
 
 
 class StripeController(http.Controller):
-    _success_url = '/payment/stripe/success'
-    _cancel_url = '/payment/stripe/cancel'
-
-    @http.route(['/payment/stripe/success', '/payment/stripe/cancel'], type='http', auth='public')
-    def stripe_success(self, **kwargs):
-        request.env['payment.transaction'].sudo().form_feedback(kwargs, 'stripe')
-        return werkzeug.utils.redirect('/payment/process')
-
-    @http.route(['/payment/stripe/s2s/create_json_3ds'], type='json', auth='public', csrf=False)
-    def stripe_s2s_create_json_3ds(self, verify_validity=False, **kwargs):
-        if not kwargs.get('partner_id'):
-            kwargs = dict(kwargs, partner_id=request.env.user.partner_id.id)
-        token = request.env['payment.acquirer'].browse(int(kwargs.get('acquirer_id'))).with_context(stripe_manual_payment=True).s2s_process(kwargs)
-
-        if not token:
-            res = {
-                'result': False,
-            }
-            return res
-
-        res = {
-            'result': True,
-            'id': token.id,
-            'short_name': token.short_name,
-            '3d_secure': False,
-            'verified': False,
+    _return_url = '/payment/stripe/return'
+
+    @http.route(_return_url, type='http', auth='public', csrf=False)
+    def stripe_success(self, **data):
+        request.env['payment.transaction'].sudo()._handle_feedback_data('stripe', data)
+        return werkzeug.utils.redirect('/payment/status')
+
+    @http.route('/payment/stripe/get_acquirer_info', type='json', auth='public')
+    def stripe_get_acquirer_info(self, acquirer_id, **kwargs):
+        # stripe_publishable_key
+        return request.env['payment.acquirer'].sudo().browse(int(acquirer_id)).stripe_publishable_key
+
+    @http.route('/payment/stripe/payment_method', type='json', auth='public', csrf=False)
+    def stripe_payment_method(self, acquirer_id, **kwargs):
+        acquirer = request.env['payment.acquirer'].sudo().browse(int(acquirer_id))
```

Comment:
```suggestion
        acquirer_sudo = request.env['payment.acquirer'].sudo().browse(acquirer_id)
```
Please specify when a record is manually sudoed to ease security analysis :)